### ISSUE TITLE
prefer-object-spread doesn't work correctly for interfaces

### ISSUE THREAD

**Issue by sebek64** [palantir/tslint#2865]
### Bug Report

- __TSLint version__: 5.4.2
- __TypeScript version__: 23.3
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
interface Dto {
};
const dto: Dto = {};
const _dto = Object.assign({}, dto);
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "prefer-object-spread": true,
  }
}
```

#### Actual behavior

This code emits a lint error:

Use the object spread operator instead.

However, after fixing to use object spread, we get a compile error:

Spread types may only be created from object types.

#### Expected behavior

tslint should not produce error in this case.

**Comment by ajafff** [comment]
I cannot reproduce the compile error: http://www.typescriptlang.org/play/#src=interface%20Dto%20%7B%0A%7D%0Aconst%20dto%3A%20Dto%20%3D%20%7B%7D%3B%0Aconst%20_dto%20%3D%20%7B%20...dto%20%7D%3B

Can you please share a code snippet that results in the behavior described above?

**Comment by sebek64** [comment]
Sorry, I've extracted insufficient code from my project, the following should work I hope:

```ts
interface Dto {
}

class X<Y extends Dto> {
    x(dto: Y) {
        const _dto = { ...dto };
    }
}
```

**Comment by ajafff** [comment]
@sebek64 
I'd say this is a bug with typescript. See https://github.com/Microsoft/TypeScript/issues/12759 and linked issue.

**Comment by sebek64** [comment]
Thank you for pointing out the bug. Yes, it is definitely a TS issue, so closing this one.
